erigon-lib/state/aggregator.go

```diff
@@ -318,7 +318,17 @@ func (a *Aggregator) OpenList(files []string, readonly bool) error {
 	return a.OpenFolder()
 }
 
+func (a *Aggregator) WaitForFiles() {
+	for {
+		select {
+		case <-a.WaitForBuildAndMerge(a.ctx):
+			return
+		}
+	}
+}
+
 func (a *Aggregator) Close() {
+	a.WaitForFiles()
```

Comment:
does it means that user can't ctrl+c erigon anymore? (merge on bor-mainnet may take 1 day)

Reply:
yes, but only in stage_custom_trace.
erigon process can be closed with ctrl-C.

I'm actually unclear if we even want this for `stage_custom_trace` -- after all we're okay with quitting between merge process -- there's protection for it. 
cc: @JkLondon

Reply:
you mean for `stage_custom_trace` , ask user "Wait for build/merge?" on ctrl-c?

Reply:
`ctrl-c` can interrupt
when execution in `stage_custom_trace` done - better wait for build/merge finish

Reply:
ah, now i understand. probably this line requires a comment - that it expecting to ctx of merge/build to be canceled.